com.unity.multiplayer.mlapi/Runtime/Messaging/InternalCommandContext.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using MLAPI.Serialization.Pooled;
+using System.Linq;
+
+namespace MLAPI.Messaging
+{
+    /// <summary>
+    /// A context used for building an internal command.
+    /// </summary>
+    internal struct InternalCommandContext : IDisposable
+    {
+        public PooledNetworkWriter NetworkWriter;
+
+        private ulong[] m_ClientIds;
+        private NetworkUpdateStage m_UpdateStage;
+        private MessageQueueContainer m_Owner;
+
+        public InternalCommandContext(PooledNetworkWriter writer, ulong[] clientIds, NetworkUpdateStage updateStage, MessageQueueContainer owner)
+        {
+            NetworkWriter = writer;
+            m_ClientIds = clientIds;
+            m_UpdateStage = updateStage;
+            m_Owner = owner;
+        }
+
+        public void Dispose()
+        {
+            Finalize();
+        }
+
+        public void Finalize()
+        {
+            if (m_Owner.NetworkManager.IsHost)
+            {
+                var containsServerClientId = m_ClientIds.Contains(m_Owner.NetworkManager.ServerClientId);
```

Comment:
Is there a way to avoid a O(n) search?  Could this be costly with a lot of connections?

Reply:
I think my serialization RFC will remove all of the special "IsHost" logic and the need to determine whether or not the server ID is included in the client list. I can try to find a better option now, if you want, but if it's not a "solve immediately" kind of issue I'd prefer to defer fixing this to when I implement that.

Reply:
Fine, your RFC eliminates this anyway I'm cool